dlc-manager/src/lib.rs

```diff
@@ -77,27 +78,79 @@ impl Time for SystemTimeProvider {
 }
 
 /// Provides signing related functionalities.
-pub trait Signer {
-    /// Signs a transaction input
-    fn sign_psbt_input(
+pub trait ContractSigner {
+    /// Get the public key associated with the [`ContractSigner`].
+    fn get_public_key(&self) -> Result<PublicKey, Error>;
+    /// Returns an arbitrary identifier describing the signer.
+    /// This should be sufficient to identify this [`ContractSigner`] object uniquely and lookup or
+    /// re-derive its keys.
+    fn contract_keys_id(&self) -> [u8; 32];
```

Comment:
What is this used for?

Reply:
oh this is from an earlier iteration i was trying, will remove